authz/grpc_audit_end2end_test.go

```diff
@@ -287,18 +308,37 @@ func TestAuditLogger(t *testing.T) {
 			stream.CloseAndRecv()
 
 			//Compare expected number of allows/denies with content of internal map of statAuditLogger
-			if lb.Stat[true] != test.wantAllows {
-				t.Errorf("Allow case failed, want %v got %v", test.wantAllows, lb.Stat[true])
+			if lb.AuthzDescisionStat[true] != test.wantAllows {
+				t.Errorf("Allow case failed, want %v got %v", test.wantAllows, lb.AuthzDescisionStat[true])
 			}
-			if lb.Stat[false] != test.wantDenies {
-				t.Errorf("Deny case failed, want %v got %v", test.wantDenies, lb.Stat[false])
+			if lb.AuthzDescisionStat[false] != test.wantDenies {
+				t.Errorf("Deny case failed, want %v got %v", test.wantDenies, lb.AuthzDescisionStat[false])
 			}
 			//Compare recorded SPIFFE Ids with the value from cert
 			for _, id := range lb.SpiffeIds {
 				if id != spiffeId {
 					t.Errorf("Unexpected SPIFFE Id, want %v got %v", spiffeId, id)
 				}
 			}
+			//Special case - compare event fields with expected values from authz policy
+			if name == `Allow All Deny Streaming - Audit All` {
+				if diff := cmp.Diff(lb.EventContent, generateEventAsMap()); diff != "" {
+					t.Fatalf("Unexpected message\ndiff (-got +want):\n%s", diff)
+				}
+			}
 		})
 	}
 }
+
+// generateEvent produces an map contaning audit.Event fields.
+// It's used to compare captured audit.Event with the matched rule during
+// `Allow All Deny Streaming - Audit All` scenario (authz_deny_all rule)
+func generateEventAsMap() map[string]string {
```

Comment:
Two alternatives to this
1. Just hard code it, you don't really need a function that returns a constant hard-coded map
2. Add `events map[string]string` as part of the table tests, then it's a part of test construction. Code it in as part of the test case, ignore it if it doesn't matter. Then you can do the `diff := cmp.Diff(...)` for this if it exists, rather than checking for a specific test name